### PR TITLE
Aligned CSS with latest Figma as of 19 March 2023

### LINKED ISSUE #138: [Super app] Bring support detail page up to date with Figma (19 Mar)
Clean up the support detail page and get UX's approval.

### PR DESCRIPTION
**Pull request by stevahnes**
Fixes #138 

### What's done?
- Also implemented the back button logic as per UX Designer's request: return to landing page.
- `SupportDetailPage` was renamed from `SupportDetailWIP`.

### What needs to be done?
- Modify the `Theme.ts` text color.
- Solve the scrolling behavior inconsistency between landing page (header also scrolls) and support detail (header is fixed).
- `SupportDetailPage` icon is not always constrained properly (see the size of IRR logo). Of course this can be easily fixed by ensuring all logos are squared up, but let's see what we can do with CSS.

### CODE REVIEW

**Comment by stevahnes** [comment]
@glencbz so sorry I messed up the previous branch commit history, so redoing it as a new branch.

Checked with Inez on the back button behavior and as per my understanding as well, it should return to previous page not the Landing Page all the time.

Also, the color for the back chevron should match that of the header text, which is why the color `sx` is `text.primary`.

**Comment by glencbz** [comment]
> @glencbz so sorry I messed up the previous branch commit history, so redoing it as a new branch.

If you're not using it, please delete the old branch.

> Checked with Inez on the back button behavior and as per my understanding as well, it should return to previous page not the Landing Page all the time.

That's okay, but did you specifically mention my concern with navigating to unrelated websites? I want to make sure she knows about this edge case before committing to a direction.

> Also, the color for the back chevron should match that of the header text, which is why the color sx is text.primary.

My comment was also that they should be the same colour, but this is the state of your PR:

image

The back chevron is clearly black, but the header is blue.

File: apps/super-app/src/routes/SupportDetailWip.tsx
Comment (glencbz) [inline apps/super-app/src/routes/SupportDetailWip.tsx:26]: I'm very sure you want `pimary.900`. This rule does nothing, the button is black.

image

See apps/super-app/src/routes/SupportDetailWip.tsx:

```
            <PublicIcon
              sx={{ color: 'primary.900', height: '1.5rem', width: '1.5rem' }}
            />
```

**Review by glencbz — CHANGES_REQUESTED** [review]

File: apps/super-app/src/routes/SupportDetailWip.tsx
Comment (glencbz) [reply apps/super-app/src/routes/SupportDetailWip.tsx:26]: Okay, I see the problem, we need to adjust `Theme.ts`.

File: apps/super-app/src/common/Theme.ts
Comment (glencbz) [inline apps/super-app/src/common/Theme.ts:1]: Notice that on L59 (where we expose the color palette to @mui), we didn't include `text.primary`, that's why `sx` doesn't pick it up.

Let's add `text: colors.text` to fix this.

File: apps/super-app/src/common/Theme.ts
Comment (glencbz) [inline apps/super-app/src/common/Theme.ts:40]: I didn't realise that these were colour constants in the file. Could we use those values instead? e.g. `colors.text.primary` instead of `'#133844`. Having magic numbers makes their meaning difficult to understand, and this will make it easier to change the theme colours if we want.

**Review by glencbz — CHANGES_REQUESTED** [review]

File: apps/super-app/src/routes/SupportDetailWip.tsx
Comment (stevahnes) [reply apps/super-app/src/routes/SupportDetailWip.tsx:26]: @glencbz ah good eye, lemme check idk why!!

**Comment by stevahnes** [comment]
@glencbz fixed! Thanks for checking the chevron color, it looked correct on my screen. Probably my screen isn't very color accurate 😢

File: apps/super-app/src/common/Theme.ts
Comment (glencbz) [inline apps/super-app/src/common/Theme.ts:47]: Nice!

File: apps/super-app/src/routes/SupportDetailWip.tsx
Comment (stevahnes) [inline apps/super-app/src/routes/SupportDetailWip.tsx:214]: @glencbz can you explain what you want to be done here again?

File: apps/super-app/src/common/components/ServiceCard.tsx
Comment (glencbz) [inline apps/super-app/src/common/components/ServiceCard.tsx:19]: Hm, maybe at this point it would be clearer to specify `margin-[top|bottom|left|right]` separately, but I don't feel strongly about it.

File: apps/super-app/src/routes/SupportDetailWip.tsx
Comment (glencbz) [inline apps/super-app/src/routes/SupportDetailWip.tsx:27]: As mentioned in the weekly meeting, I think we can merge this PR with `navigate(-1)` if we leave a TODO to check the edge cases with UXers.

File: apps/super-app/src/routes/SupportDetailWip.tsx
Comment (stevahnes) [reply apps/super-app/src/routes/SupportDetailWip.tsx]: Left a TODO as discussed.

File: apps/super-app/src/routes/SupportDetailWip.tsx
Comment (stevahnes) [reply apps/super-app/src/routes/SupportDetailWip.tsx:28]: Left a TODO as discussed.

File: apps/super-app/src/common/components/ServiceCard.tsx
Comment (stevahnes) [reply apps/super-app/src/common/components/ServiceCard.tsx]: Hmmm, feels like it'll be too verbose. Shall we leave it as-is for now?

Comment (glencbz) [reply apps/super-app/src/common/components/ServiceCard.tsx]: IMO verbosity is less of a problem than understandability, but let's leave it.

File: apps/super-app/src/routes/SupportDetailPage.tsx
Comment (glencbz) [inline apps/super-app/src/routes/SupportDetailPage.tsx:105]: What do you mean by this, and how will we know when we've fixed it?

File: apps/super-app/src/routes/SupportDetailPage.tsx
Comment (glencbz) [inline apps/super-app/src/routes/SupportDetailPage.tsx:149]: Is this stale? I see that we're already parameterising on `ctaLogo`.

File: apps/super-app/src/routes/SupportDetailPage.tsx
Comment (glencbz) [inline apps/super-app/src/routes/SupportDetailPage.tsx:171]: This looks unrelated from the CSS. Can we remove these changes?

File: apps/super-app/src/services/DetailServices.ts
Comment (glencbz) [inline apps/super-app/src/services/DetailServices.ts:38]: Maybe `ctaIcon`, since it's more of a decorative icon than a logo?

File: apps/super-app/src/services/DetailServices.ts
Comment (glencbz) [inline apps/super-app/src/services/DetailServices.ts:110]: I strongly feel that this is change brings us in the wrong direction:

a) We want to replace these hardcoded values with Strapi eventually, but this function doesn't emulate a Strapi calling pattern at all. This takes a single string as a parameter and returns 1 of 3 values we care about, but we plan to use Strapi single types for each page, where a single type has its own url and set of properties.
b) It's replacing a declarative, easy to read style with imperative assignments

**Review by glencbz — CHANGES_REQUESTED** [review]
It looks like you might have accidentally included other changes into this PR. Do send them as a separate PR if this was intentional.

File: apps/super-app/src/routes/SupportDetailPage.tsx
Comment (glencbz) [inline apps/super-app/src/routes/SupportDetailPage.tsx:1]: I think a better direction is to turn this into the TWC2 page. Each support page is so different from one another that we probably can't reuse a single page for all of them anyway.

This was what I meant by `TWC2-only` in the PR description in https://github.com/bettersg/call-home/pull/115, and `move each page's things into their own files` in the commit message of https://github.com/bettersg/call-home/pull/115/commits/37b0d2f4be7e3af5824e4552c261bcc24752c93c

File: apps/super-app/src/routes/SupportDetailPage.tsx
Comment (stevahnes) [reply apps/super-app/src/routes/SupportDetailPage.tsx:105]: @glencbz I discovered it when I tried adding the IRR logo into the Details Page. It blew up instead of being constrained to the max-height. Likely coz the logo was tall aspect ratio wise.

We can simply square all our logos to circumvent this actually.

File: apps/super-app/src/common/Theme.ts
Comment (stevahnes) [reply apps/super-app/src/common/Theme.ts:68]: I don't get this, I tried and it doesn't work. Also added `main` field into the `colors` object but it doesn't work either.

Comment (glencbz) [reply apps/super-app/src/common/Theme.ts:68]: What did you try specifically? I changed this to

```
      text: colors.text,
```

and it compiles just fine for me. I'd expect it to work for you too, because `colors.text`(what I used) and `{ primary: colors.text.primary }` (what you used in this PR) have exactly the same type.

File: apps/super-app/src/routes/SupportDetailPage.tsx
Comment (glencbz) [inline apps/super-app/src/routes/SupportDetailPage.tsx:147]: If possible, do leave your own interpretation of the conversation so that a) readers don't have to make a round trip to reread the conversation b) readers know exactly what you're referencing instead of reading the whole conversation to figure it out.

In this case, I'm not sure what you're referencing because you've already implemented my suggestion here:

> If we want a generic CTA that can accept any image URL, then sure, parameterising is fine

by accepting the `ctaIcon` prop, so I don't think there's anything left to do.

File: apps/super-app/src/services/DetailServices.ts
Comment (glencbz) [inline apps/super-app/src/services/DetailServices.ts:9]: This renaming from `ServiceCardAbout` to `ServiceCardDetail` looks like a leftover from the previous PR iteration. I'd prefer to undo this, but I suppose the name makes more sense, so we can keep it if we want.

File: apps/super-app/src/routes/index.tsx
Comment (glencbz) [inline apps/super-app/src/routes/index.tsx:3]: Since we aren't using a generic `SupportDetailPage` any more, perhaps we could omit this change for now and reintroduce it when it makes sense.

File: apps/super-app/src/routes/SupportDetailPage.tsx
Comment (glencbz) [inline apps/super-app/src/routes/SupportDetailPage.tsx:28]: Did we change course on the navigation?

File: apps/super-app/src/routes/SupportDetailPage.tsx
Comment (glencbz) [reply apps/super-app/src/routes/SupportDetailPage.tsx]: That works, though I think there is probably an elegant CSS-based solution for this (e.g. setting the `height` of the containing element). We don't have to discuss this now; we can defer this work until we need it.

**Review by glencbz — CHANGES_REQUESTED** [review]
Thanks! I only have small comments this time, let's clean those up before merging.

Also, it looks like the PR description is out of date. If you could update that to reflect the current reality, that would greatly help me understand this PR :)

File: apps/super-app/src/routes/SupportDetailPage.tsx
Comment (stevahnes) [reply apps/super-app/src/routes/SupportDetailPage.tsx:28]: Yes we did. There's a new mock. Let's take the alignment for this one later, but get the logic out first.

File: apps/super-app/src/common/Theme.ts
Comment (stevahnes) [reply apps/super-app/src/common/Theme.ts:68]: image

File: apps/super-app/src/services/DetailServices.ts
Comment (stevahnes) [reply apps/super-app/src/services/DetailServices.ts:9]: Oh no, this one is deliberate as I also feel this name makes more sense :)

File: apps/super-app/src/common/Theme.ts
Comment (glencbz) [reply apps/super-app/src/common/Theme.ts:68]: I see, so you had:

```
text: {
  primary: colors.text
}
```

but what I meant was:

```
text: colors.text
```

File: apps/super-app/src/routes/index.tsx
Comment (glencbz) [reply apps/super-app/src/routes/index.tsx:3]: >  If we are going to have individual pages, we can rename them to specific providers later?

Yes. I meant that we would rename them to e.g. `SupportTwc2`.

> I just feel that the "WIP" term is misleading. It's not really "WIP" when you finish one page, no?

Yes, we'll get rid of it when we are done by renaming to `SupportTwc2`. If we are done now, we can make that change.

File: apps/super-app/src/services/DetailServices.ts
Comment (glencbz) [reply apps/super-app/src/services/DetailServices.ts:10]: Sure. In the future, it would be helpful to call out intentional changes in the PR message or in a comment.

**Review by glencbz — APPROVED** [review]
Actually, let's just merge this and clean up the `text:` thing later. It's not big enough to be worth holding up this PR. Sorry for the delay.

As usual, squash and merge would be greatly appreciated.

**Pull request merged**
